src/actions.rs

```diff
@@ -478,14 +478,10 @@ impl ActionHandler {
         let mut buf = Vec::<u8>::new();
         match format_input(input, &config, Some(&mut buf)) {
             Ok((summary, ..)) => {
-                // It's workaround because currently rustfmt returns ok even if there are parsing errors.
-                // If there are parsing errors the buffer is empty.
-                // Without this condition a text document is erased.
-                if summary.has_parsing_errors() {
-                    debug!("reformat: format_input failed: has parsing errors");
-
-                    out.failure(id, "Reformat failed to complete successfully")
-                } else {
+                // format_input returns Ok even if there are any errors, i.e., parsing errors.
+                // If there are any errors the buffer is empty.
```

Comment:
@nrc,
My comment explains why the code is written like that.
Did you mean "If there are any errors the buffer is empty"?
If yes, then it was made to make clear the next paragraph about "An empty buffer".